workspaces/vscode-python/Dockerfile

```diff
@@ -0,0 +1,140 @@
+# Enhanced code-server workspace template for PrairieLearn.
+# As discussed here: https://github.com/PrairieLearn/PrairieLearn/issues/3170
+
+FROM --platform=linux/amd64 codercom/code-server:4.20.1-bookworm
+
+# On PL, we want to standardize on using 1001:1001 for the user. We change the
+# "coder" account's UID and GID here so that fixuid has no work to do later.
+# This speeds up container loading time drastically and avoids timeouts.
+USER root
+RUN export OLD_UID=$(id -u coder) && \
+	export OLD_GID=$(id -g coder) && \
+	export NEW_UID=1001 && \
+	export NEW_GID=1001 && \
+	groupmod -g 1001 coder && \
+	usermod -u 1001 -g 1001 coder && \
+	find /home -user $OLD_UID -execdir chown -h $NEW_UID {} + && \
+	find /home -group $OLD_GID -execdir chgrp -h $NEW_GID {} + && \
+	unset OLD_UID OLD_GID NEW_UID NEW_GID
```

Comment:
You did need it previously but weren't doing it, which is why I made the hacked template and began sharing it with staff directly (225 and others are using it). It's needed because the image assumes running as 1000:1000 but PL runs as 1001:1001. The code-server image can try to chown a bunch of stuff at runtime to band-aid the issue, but that is so slow that it times out the startup (225 had this problem specifically).

Reply:
That's helpful context! I'm happy to include it if it's necessary, just wanted to make sure I understand why it is.

Tangent: does Jupyterlab require (or would it benefit from) similar treatment?

Reply:
Jupyter may indeed benefit from similar treatment, although I think they also use something to automatically chown at runtime. So making this change ahead of time is mainly a startup latency benefit.